### PR TITLE
Update tasks and 🛳️  mask-generation and zero-shot-object-detection

### PR DESCRIPTION
**Pull request by merveenoyan**
This is a follow-up to my previous PR to add resources to various tasks and ship mask-generation and zero-shot-object-detection

### CODE REVIEW

File: packages/tasks/src/tasks/text-generation/about.md [hunk 1]
```diff
@@ -110,25 +110,39 @@ Would you like to learn more about the topic? Awesome! Here you can find some cu
...
 
-### Course and Blogs
 
-- [Course Chapter on Training a causal language model from scratch](https://huggingface.co/course/chapter7/6?fw=pt)
-- [TO Discussion with Victor Sanh](https://www.youtube.com/watch?v=Oy49SCW_Xpw&ab_channel=HuggingFace)
-- [Hugging Face Course Workshops: Pretraining Language Models & CodeParrot](https://www.youtube.com/watch?v=ExUR7w6xe94&ab_channel=HuggingFace)
-- [Training CodeParrot 🦜 from Scratch](https://huggingface.co/blog/codeparrot)
-- [How to generate text: using different decoding methods for language generation with Transformers](https://huggingface.co/blog/how-to-generate)
+### Model Inference & Deployment
+- [Optimizing your LLM in production](https://huggingface.co/blog/optimize-llm)
```
Comment (osanseviero) [inline packages/tasks/src/tasks/text-generation/about.md:117]: Let's leave a new line between headings and lists, as some editors won't work well otherwise

File: packages/tasks/src/tasks/text-generation/about.md [hunk 2]
```diff
@@ -110,25 +110,39 @@ Would you like to learn more about the topic? Awesome! Here you can find some cu
...
+
+### Model Fine-tuning/Training
+- [Non-engineers guide: Train a LLaMA 2 chatbot](https://huggingface.co/blog/Llama2-for-non-engineers)
+- [Training CodeParrot 🦜 from Scratch](https://huggingface.co/blog/codeparrot)
 - [Creating a Coding Assistant with StarCoder](https://huggingface.co/blog/starchat-alpha)
+
+### Advanced Concepts Explained Simply
+- [Mixture of Experts Explained](https://huggingface.co/blog/moe)
+
+### Advanced Fine-tuning/Training Recipes
```
Comment (osanseviero) [inline packages/tasks/src/tasks/text-generation/about.md:135]:
```suggestion
### Advanced Fine-tuning/Training Recipes
```

File: packages/tasks/src/tasks/text-generation/about.md [hunk 3]
```diff
@@ -110,25 +110,39 @@ Would you like to learn more about the topic? Awesome! Here you can find some cu
...
+- [Fine-tuning Llama 2 70B using PyTorch FSDP](https://huggingface.co/blog/ram-efficient-pytorch-fsdp)
+- [The N Implementation Details of RLHF with PPO](https://huggingface.co/blog/the_n_implementation_details_of_rlhf_with_ppo)
+- [Preference Tuning LLMs with Direct Preference Optimization Methods](https://huggingface.co/blog/pref-tuning)
+- [Fine-tune Llama 2 with DPO](https://huggingface.co/blog/dpo-trl)
+
+### Hugging Face Model Releases
```
Comment (osanseviero) [inline packages/tasks/src/tasks/text-generation/about.md:141]: I would not have a section about HF-specific models (also IDEFICS is multimodal)

File: packages/tasks/src/tasks/text-to-image/about.md [hunk 4]
```diff
@@ -53,14 +53,21 @@ await inference.textToImage({
...
-- [MinImagen - Build Your Own Imagen Text-to-Image Model](https://www.assemblyai.com/blog/minimagen-build-your-own-imagen-text-to-image-model/)
-- [Using LoRA for Efficient Stable Diffusion Fine-Tuning](https://huggingface.co/blog/lora)
 - [Using Stable Diffusion with Core ML on Apple Silicon](https://huggingface.co/blog/diffusers-coreml)
 - [A guide on Vector Quantized Diffusion](https://huggingface.co/blog/vq-diffusion)
 - [🧨 Stable Diffusion in JAX/Flax](https://huggingface.co/blog/stable_diffusion_jax)
 - [Running IF with 🧨 diffusers on a Free Tier Google Colab](https://huggingface.co/blog/if)
+- [Introducing Würstchen: Fast Diffusion for Image Generation](https://huggingface.co/blog/wuerstchen)
+- [Efficient Controllable Generation for SDXL with T2I-Adapters](https://huggingface.co/blog/t2i-sdxl-adapters)
+
+### Model Fine-tuning
```
Comment (osanseviero) [inline packages/tasks/src/tasks/text-to-image/about.md:68]:
```suggestion
### Model Fine-tuning
```

**Review by osanseviero — COMMENTED** [review]
Looking nice!

**Comment by merveenoyan** [comment]
I will lint after the last review to the content

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 5]
```diff
@@ -6,25 +6,53 @@ When filtering for an image, the generated masks might serve as an initial filte
...
-outputs = generator(image_url, points_per_batch = 256)
+outputs = generator(image_url, points_per_batch = 64)
+outputs["masks"]
+# array of multiple binary masks returned for each generated mask
+```
+
+The prompt mode takes in three types of prompts.
+- **Point prompt:** The user can leave a point on the image, and a meaningful segment around the point will be returned.
+- **Box prompt:** The user can draw a box on the image, and a meaningful segment within the box will be returned.
+- **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability has not yet been released and has only been explored in research.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/about.md:39]:
```suggestion
- **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability is not available in all models and has only been explored in research.
```

If no model supports it yet maybe we should not list it?

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 6]
```diff
@@ -6,25 +6,53 @@ When filtering for an image, the generated masks might serve as an initial filte
...
+
+Below you can see how to input point prompting with model inference without `pipeline`.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/about.md:41]:
```suggestion
Below you can see how to use an input-point prompt. It also demonstrates direct model inference without the `pipeline` abstraction.

```

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 7]
```diff
@@ -6,25 +6,53 @@ When filtering for an image, the generated masks might serve as an initial filte
...
+Below you can see how to input point prompting with model inference without `pipeline`. 
+```python
+from transformers import SamModel, SamProcessor
+from PIL import Image
+import requests 
+
+model = SamModel.from_pretrained("Zigeng/SlimSAM-uniform-50").to("cuda")
+processor = SamProcessor.from_pretrained("Zigeng/SlimSAM-uniform-50")
+
+raw_image = Image.open(requests.get(image_url, stream=True).raw).convert("RGB")
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/about.md:50]: maybe repeat the url here, so the example is self-contained?

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 8]
```diff
@@ -6,25 +6,53 @@ When filtering for an image, the generated masks might serve as an initial filte
...
+input_points = [[[450, 600]]]
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/about.md:51]: maybe briefly explain the shape? (bs, number_of_points, 2 (coords per point)). Also, does the processor reshape to include the batch size if we just pass a list of points?

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 9]
```diff
@@ -6,25 +6,53 @@ When filtering for an image, the generated masks might serve as an initial filte
...
+inputs = processor(raw_image, input_points=input_points, return_tensors="pt").to("cuda")
+outputs = model(**inputs)
+masks = processor.image_processor.post_process_masks(outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu())
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/about.md:54]: Is there a direct path from the processor to `post_process_masks`?

File: packages/tasks/src/tasks/text-classification/about.md [hunk 10]
```diff
@@ -150,6 +150,7 @@ classifier("I will walk to home when I went through the bus.")
 
 Would you like to learn more about the topic? Awesome! Here you can find some curated resources that you may find helpful!
 
+- [SetFitABSA: Few-Shot Aspect Based Sentiment Analysis using SetFit](https://huggingface.co/blog/setfit-absa)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-classification/about.md:153]: This is a great resource, but maybe a bit advanced to place it the first in the list (not sure)

File: packages/tasks/src/tasks/text-generation/about.md [hunk 11]
```diff
@@ -110,25 +110,35 @@ Would you like to learn more about the topic? Awesome! Here you can find some cu
 - [ChatUI Docker Spaces](https://huggingface.co/docs/hub/spaces-sdks-docker-chatui)
 - [Causal language modeling task guide](https://huggingface.co/docs/transformers/tasks/language_modeling)
 - [Text generation strategies](https://huggingface.co/docs/transformers/generation_strategies)
+- [Course Chapter on Training a causal language model from scratch](https://huggingface.co/course/chapter7/6?fw=pt)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-generation/about.md:113]:
```suggestion
- [Course chapter on training a causal language model from scratch](https://huggingface.co/course/chapter7/6?fw=pt)
```

(Or title casing, but using it consistently :))

File: packages/tasks/src/tasks/text-to-image/about.md [hunk 12]
```diff
@@ -53,14 +53,21 @@ await inference.textToImage({
 
 ## Useful Resources
 
+### Model Inference
 - [Hugging Face Diffusion Models Course](https://github.com/huggingface/diffusion-models-class)
+- [Welcome aMUSEd: Efficient Text-to-Image Generation](https://huggingface.co/blog/amused)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-to-image/about.md:58]: This is a bit experimental, I think the diffusers guides (at least) should come first.

File: packages/tasks/src/tasks/text-to-image/about.md [hunk 13]
```diff
@@ -53,14 +53,21 @@ await inference.textToImage({
 
 ## Useful Resources
 
+### Model Inference
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-to-image/about.md:56]:
```suggestion
### Model Inference

```

File: packages/tasks/src/tasks/text-to-image/about.md [hunk 14]
```diff
@@ -53,14 +53,21 @@ await inference.textToImage({
...
-- [MinImagen - Build Your Own Imagen Text-to-Image Model](https://www.assemblyai.com/blog/minimagen-build-your-own-imagen-text-to-image-model/)
-- [Using LoRA for Efficient Stable Diffusion Fine-Tuning](https://huggingface.co/blog/lora)
 - [Using Stable Diffusion with Core ML on Apple Silicon](https://huggingface.co/blog/diffusers-coreml)
 - [A guide on Vector Quantized Diffusion](https://huggingface.co/blog/vq-diffusion)
 - [🧨 Stable Diffusion in JAX/Flax](https://huggingface.co/blog/stable_diffusion_jax)
 - [Running IF with 🧨 diffusers on a Free Tier Google Colab](https://huggingface.co/blog/if)
+- [Introducing Würstchen: Fast Diffusion for Image Generation](https://huggingface.co/blog/wuerstchen)
+- [Efficient Controllable Generation for SDXL with T2I-Adapters](https://huggingface.co/blog/t2i-sdxl-adapters)
+
+### Model Fine-tuning
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-to-image/about.md:68]:
```suggestion
### Model Fine-tuning

```

File: packages/tasks/src/tasks/text-to-image/about.md [hunk 15]
```diff
@@ -53,14 +53,21 @@ await inference.textToImage({
...
+- [Finetune Stable Diffusion Models with DDPO via TRL](https://huggingface.co/blog/pref-tuning)
+- [LoRA training scripts of the world, unite!](https://huggingface.co/blog/sdxl_lora_advanced_script)
+- [Using LoRA for Efficient Stable Diffusion Fine-Tuning](https://huggingface.co/blog/lora)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-to-image/about.md:71]: I'd put them exactly in reverse order lol. Also https://huggingface.co/blog/annotated-diffusion is a great, great resource (and not only for fine-tuning).

File: packages/tasks/src/tasks/text-to-speech/about.md [hunk 16]
```diff
@@ -61,3 +61,5 @@ await inference.textToSpeech({
 - [An introduction to SpeechT5, a multi-purpose speech recognition and synthesis model](https://huggingface.co/blog/speecht5).
 - [A guide on Fine-tuning Whisper For Multilingual ASR with 🤗Transformers](https://huggingface.co/blog/fine-tune-whisper)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-to-speech/about.md:62]: this is not tts, is it?

File: packages/tasks/src/tasks/text-to-speech/about.md [hunk 17]
```diff
@@ -61,3 +61,5 @@ await inference.textToSpeech({
...
 - [Speech Synthesis, Recognition, and More With SpeechT5](https://huggingface.co/blog/speecht5)
+- [Optimizing a Text-To-Speech model using 🤗 Transformers](https://huggingface.co/blog/optimizing-bark)
+-
```
Comment (pcuenca) [inline packages/tasks/src/tasks/text-to-speech/about.md:65]: cc @Vaibhavs10 in case there are some obvious tts resources we can use here :)

**Review by pcuenca — COMMENTED** [review]
I focused on details mostly, but it looks good!

File: packages/tasks/src/tasks/text-to-image/about.md (hunk 15 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/text-to-image/about.md:73]: They weren't in particular order honestly

File: packages/tasks/src/tasks/mask-generation/about.md (hunk 9 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/mask-generation/about.md:55]: sorry this is my fault I think you're right

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 18]
```diff
@@ -6,25 +6,56 @@ When filtering for an image, the generated masks might serve as an initial filte
...
+# array of multiple binary masks returned for each generated mask
+```
+
+Prompt mode takes in three types of prompts:
+
+- **Point prompt:** The user can select a point on the image, and a meaningful segment around the point will be returned.
+- **Box prompt:** The user can draw a box on the image, and a meaningful segment within the box will be returned.
+- **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability has not yet been released and has only been explored in research.
+
+Below you can see how to use an input-point prompt. It also demonstrates direct model inference without the `pipeline` abstraction. The input prompt here is a list of lists where outermost list is number of batches, inner one is number of points and innermost one is the actual location of the point.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/mask-generation/about.md:42]:
```suggestion
Below you can see how to use an input-point prompt. It also demonstrates direct model inference without the `pipeline` abstraction. The input prompt here is a nested list where the outermost list is the batch size (`1`), then the number of points (also `1` in this example), and the innermost list contains the actual coordinates of the point (`[450, 600]`).
```

**Review by pcuenca — APPROVED** [review]
